### PR TITLE
#3266: Add additional modules for configuration staging workflow(s).

### PR DESCRIPTION
**Pull request by joeparsons**
## Description

Adds additional modules (as optional composer dependencies) that are typically used by sites employing some kind of configuration staging workflow.

Associated Pantheon upstream repo PR:
https://github.com/az-digital/az-quickstart-pantheon/pull/223

## Related issues

#3266

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [x] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: config/optional/config_ignore.settings.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+mode: simple
+ignored_config_entities:
+  - 'block.block.*'
+  - exclude_node_title.settings
+  - 'system.menu.*'
+  - system.site
+  - 'taxonomy.vocabulary.*'
+  - 'webform.webform.*'
```
Comment (joeparsons) [inline config/optional/config_ignore.settings.yml:8]: A more complete list of configs that content admins/editors have access to would look like this.  We need to discuss whether we should ignore all of these or not.
```suggestion
  - 'block.block.*'
  - block_class.settings
  - easy_breadcrumb.settings
  - exclude_node_title.settings
  - 'language.content_settings.*.*'
  - 'language.entity.*'
  - 'metatag.metatag_defaults.*'
  - 'shortcut.set.*'
  - 'system.menu.*'
  - system.site
  - 'taxonomy.vocabulary.*'
  - 'webform.webform.*'
  - xmlsitemap.settings
  - 'xmlsitemap.settings.*.*'
  - 'xmlsitemap.xmlsitemap.*'
```

File: .probo.yaml [hunk 2]
```diff
@@ -70,6 +70,7 @@ steps:
       - $SRC_DIR/az-quickstart-scaffolding/vendor/bin/drush -y -n pm:install environment_indicator
       - chmod 777 /var/www/html/sites/default/settings.php
       - echo -e "\$config['environment_indicator.indicator']['name'] = '$BRANCH_NAME (ProboCI)';\n\$config['environment_indicator.indicator']['bg_color'] = '#440027';\n\$config['environment_indicator.indicator']['fg_color'] = '#ffffff';" >> /var/www/html/sites/default/settings.php
+      - echo -e "\$settings['config_readonly'] = TRUE;" >> /var/www/html/sites/default/settings.php
```
Comment (joeparsons) [inline .probo.yaml:73]: We'll probably want to remove this before merging.

Comment (joeparsons) [inline .probo.yaml:73]:
```suggestion
```

**Comment by joeparsons** [comment]
Since I added the the default configuration settings for `config_ignore` to `config/optional` in this PR, I see a PHP error on the main core configuration synchronization UI.  This seems to be tied to no config being exported in the config sync directory.

Reported an issue in the Config Ignore issue queue here: https://www.drupal.org/project/config_ignore/issues/3432315

**Comment by joeparsons** [comment]
## From 2023-03-20 workshop discussion

- There may be additional configs we want to ignore as well.  Example from Troy: environment-specific config related to email settings.

File: config/optional/config_ignore.settings.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+mode: simple
+ignored_config_entities:
+  - 'block.block.*'
```
Comment (joeparsons) [inline config/optional/config_ignore.settings.yml:3]:
```suggestion
  - az_core.settings
  - az_enterprise_attributes_import.settings
  - az_event_trellis.settings
  - az_publication.settings
  - 'block.block.*'
```

File: composer.json [hunk 4]
```diff
@@ -160,6 +164,9 @@
             "drupal/config_distro": {
                 "Storage comparer issue": "https://gist.githubusercontent.com/tadean/e2e0dd57f1a9c4eae28d8e713cb01d82/raw/e114987d4a76c183208c763de4dbfecbf41b7f35/config_distro_storage_comparer_issue.patch"
             },
+            "drupal/config_ignore": {
+                "Ignoring any config entities with an empty config sync directory causes PHP erro (3432315)": "https://git.drupalcode.org/project/config_ignore/-/merge_requests/25.diff"
```
Comment (joeparsons) [inline composer.json:168]:
```suggestion
                "Ignoring any config entities with an empty config sync directory causes PHP erro (3432315)": "https://www.drupal.org/files/issues/2024-04-03/3432315-13.patch"
```

**Review by tadean — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -160,6 +164,9 @@
             "drupal/config_distro": {
                 "Storage comparer issue": "https://gist.githubusercontent.com/tadean/e2e0dd57f1a9c4eae28d8e713cb01d82/raw/e114987d4a76c183208c763de4dbfecbf41b7f35/config_distro_storage_comparer_issue.patch"
             },
+            "drupal/config_ignore": {
+                "Ignoring any config entities with an empty config sync directory causes PHP erro (3432315)": "https://www.drupal.org/files/issues/2024-04-03/3432315-13.patch"
```
Comment (trackleft) [inline composer.json:168]:
```suggestion
                "Ignoring any config entities with an empty config sync directory causes PHP error (3432315)": "https://www.drupal.org/files/issues/2024-04-03/3432315-13.patch"
```

**Review by tadean — APPROVED** [review]

**Pull request merged**